### PR TITLE
[spec] Remove fallback, per 2018.11.28 TC39 feedback

### PR DESCRIPTION
**Pull request by ljharb**
Closes #39.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -9,10 +9,8 @@ When the `matchAll` method is called, the following steps are taken:
     1. If *matcher* is not **undefined**, then
       1. Return ? [Call](call)(*matcher*, *regexp*, &laquo; *O* &raquo;).
   1. Let *S* be ? [ToString][tostring](*O*).
-  1. Let *matcher* be ? [RegExpCreate][regexp-create](*R*, `"g"`).
-  1. Let *global* be **true**.
-  1. Let *fullUnicode* be **false**.
-  1. Return ! [CreateRegExpStringIterator](#createregexpstringiterator-abstract-operation)(*matcher*, *S*, *global*, *fullUnicode*).
+  1. Let *rx* be ? [RegExpCreate][regexp-create](*R*, `"g"`).
```
Comment (mathiasbynens) [inline spec.md]: `*R*` → `*regexp*`

**Comment by mathiasbynens** [comment]
I love how close this is to https://tc39.github.io/ecma262/#sec-string.prototype.match now!

File: spec.md (hunk 1 above)
Comment (ljharb) [reply spec.md]: thanks, looks like this bug predates this PR :-D

**Comment by ljharb** [comment]
cc @anba

File: spec.emu [hunk 2]
```diff
@@ -21,10 +21,8 @@ contributors: Jordan Harband
 			1. If _matcher_ is not *undefined*, then
 				1. Return ? Call(_matcher_, _regexp_, &laquo; _O_ &raquo;).
 		1. Let _S_ be ? ToString(_O_).
-		1. Let _matcher_ be ? RegExpCreate(_R_, `"g"`).
-		1. Let _global_ be *true*.
-		1. Let _fullUnicode_ be *false*.
-		1. Return ! CreateRegExpStringIterator(_matcher_, _S_, _global_, _fullUnicode_).
+		1. Let _rx_ be ? RegExpCreate(_regexp_, `"g"`).
+		1. Return ? Invoke(_rx_, @@matchAll, &laquo; _O_ &raquo;).
```
Comment (zenparsing) [inline spec.emu]: Should this be `_S_` instead of `_O_`?

Comment (ljharb) [reply spec.emu]: indeed it should! thanks, good catch.

**Comment by anba** [comment]
LGTM!

**Comment by peterwmwong** [comment]
@ljharb @bterlson is there anything holding up this PR from being approved/merged?
Thanks.

**Pull request merged**